### PR TITLE
Show total invested vCow amount

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

Calculates and shows total amount of invested vCow on the InvestFlowStep 1
Screenshot from 2022-01-19 19-59-31

### CODE REVIEW

File: src/custom/state/claim/hooks/index.ts [hunk 1]
```diff
@@ -849,3 +849,28 @@ function _getPrice({ token, amount }: { amount: string; token: Token | GpEther }
     quoteAmount: CurrencyAmount.fromRawAmount(token, amount),
   }).invert()
 }
+
+export function useTotalVCowAmount() {
+  const { selected, activeClaimAccount, investFlowData } = useClaimState()
+  const claims = useUserEnhancedClaimData(activeClaimAccount)
+
+  const zeroVCow = CurrencyAmount.fromRawAmount(V_COW[SupportedChainId.RINKEBY], '0')
```
Comment (anxolin) [inline src/custom/state/claim/hooks/index.ts:857]: Why hardcoded to Rinkeby?

File: src/custom/state/claim/hooks/index.ts [hunk 2]
```diff
@@ -849,3 +849,28 @@ function _getPrice({ token, amount }: { amount: string; token: Token | GpEther }
...
+
+  return claims.reduce<typeof zeroVCow>((acc, claim) => {
```
Comment (anxolin) [inline src/custom/state/claim/hooks/index.ts:859]: this code could use some comments to make it easier to read

File: src/custom/state/claim/hooks/index.ts [hunk 3]
```diff
@@ -849,3 +849,28 @@ function _getPrice({ token, amount }: { amount: string; token: Token | GpEther }
...
+
+      if (!investedAmount || !price || !currencyAmount) {
+        return acc
+      }
+
+      const investA = CurrencyAmount.fromRawAmount(currencyAmount.currency, investedAmount)
+      const vCowAmount = price.quote(investA)
+      return acc.add(vCowAmount as CurrencyAmount<Token>)
+    } else {
+      return acc
```
Comment (anxolin) [inline src/custom/state/claim/hooks/index.ts:873]: ubernit: we could call acc something like `totalTokens` or something to make it easier to read?

File: src/custom/state/claim/hooks/index.ts [hunk 4]
```diff
@@ -849,3 +849,28 @@ function _getPrice({ token, amount }: { amount: string; token: Token | GpEther }
...
+    const { price, currencyAmount } = claim
+    if (selected.includes(claim.index)) {
+      const investClaim = investFlowData.find(({ index }) => index === claim.index)
+      const investedAmount = investClaim?.investedAmount
+
+      if (!investedAmount || !price || !currencyAmount) {
+        return acc
+      }
+
+      const investA = CurrencyAmount.fromRawAmount(currencyAmount.currency, investedAmount)
```
Comment (anxolin) [inline src/custom/state/claim/hooks/index.ts:869]: maybe find a better name for `investA`

**Review by anxolin — COMMENTED** [review]
Glad you got what we were meaning with doing a hook that would wrap `useUserEnhancedClaimData`

File: src/custom/state/claim/hooks/index.ts [hunk 5]
```diff
@@ -849,3 +849,36 @@ function _getPrice({ token, amount }: { amount: string; token: Token | GpEther }
...
+  if (!chainId) {
+    return undefined
+  }
+
+  const zeroVCow = CurrencyAmount.fromRawAmount(V_COW[chainId], '0')
+
+  return claims.reduce<typeof zeroVCow>((totalTokens, claim) => {
+    const { price, currencyAmount } = claim
+    // if claim is included in selected invest option
+    if (selected.includes(claim.index)) {
```
Comment (alfetopito) [inline src/custom/state/claim/hooks/index.ts:867]: What about free claims?

Comment (nenadV91) [reply src/custom/state/claim/hooks/index.ts:867]: Based on text and place, I assume this is only to show investment claims sum `vCOW to receive based on selected investment(s):`

**Comment by alfetopito** [comment]
While implementing https://github.com/gnosis/cowswap/pull/2227 I think this is no longer relevant.

If we take the route we discussed on Figma, we won't show this value on the allowances page (without the free claims).
Instead we'll show it on the review page, where all the data is already present.
Will just need to sum it up there directly, so this hook will be redundant.

**Comment by elena-zh** [comment]
I get a hard error when try to open the Claim page as a not connected user/connected to Mainnet/xDAI user
image. 
When I'm connected to Rinkeby, I can open the Claim page

**Review by elena-zh — CHANGES_REQUESTED** [review]
Besides mentioned above issue, calculation itself works fine to me

**Comment by nenadV91** [comment]
@elena-zh Thank you Elena, should be fixed now.

**Comment by anxolin** [comment]
> If we take the route we discussed on Figma, we won't show this value on the allowances page (without the free claims).
Instead we'll show it on the review page, where all the data is already present.
Will just need to sum it up there directly, so this hook will be redundant.

Yes, this is also what we discussed in the last sync. I tried to suggest this in my doc of outstanding things and this message https://gnosisinc.slack.com/archives/C025G521XQD/p1642549623080200

Happy to remove the totals in the approvals page

**Review by anxolin — COMMENTED** [review]
> If we take the route we discussed on Figma, we won't show this value on the allowances page (without the free claims).
Instead we'll show it on the review page, where all the data is already present.
Will just need to sum it up there directly, so this hook will be redundant.

Yes, this is also what we discussed in the last sync. I tried to suggest this in my doc of outstanding things and this message https://gnosisinc.slack.com/archives/C025G521XQD/p1642549623080200

Happy to remove the totals in the approvals page

**Comment by elena-zh** [comment]
Thank you, @nenadV91 !
The claim page is displayed now when I open it as a not connected user/user connect to Mainnet/xdai

**Pull request closed without merging**